pkg/webui/lib/hooks/use-bookmark.js

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2024 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useCallback } from 'react'
+
+import { getApplication } from '@console/store/actions/applications'
+import { getGateway } from '@console/store/actions/gateways'
+import { getOrganization } from '@console/store/actions/organizations'
+import { getUser } from '@account/store/actions/user'
+import { getDevice } from '@console/store/actions/devices'
+import { getClient } from '@account/store/actions/clients'
+
+const iconMap = {
+  application: 'application',
+  gateway: 'gateway',
+  organization: 'organization',
+  user: 'user',
+  client: 'client',
+  device: 'device',
+}
+
+const getApplicationName = (appId, selectors) => getApplication(appId.id, selectors)
+const getGatewayName = (gatewayId, selectors) => getGateway(gatewayId.id, selectors)
+const getOrganizationName = (organizationId, selectors) =>
+  getOrganization(organizationId.id, selectors)
+const getUserName = (userId, selectors) => getUser(userId.id, selectors)
+const getClientName = (clientId, selectors) => getClient(clientId.id, selectors)
+const getDeviceName = (devId, selectors) => getDevice(devId.appId, devId.id, selectors)
```

Comment:
Is this whole wrapping needed? Couldn't you just invoke the action creator directly, by passing the ID string.